### PR TITLE
Build docs and upload to gh-pages

### PR DESCRIPTION
**Pull request by godexsoft**
For #889 

This is still WIP.
The first step was to fix cmake so that generating documentation did not depend on Conan (and all Clio's deps).
Now we have a new workflow that we need to get to work but in this PR it will always fail because the permissions were missing and will only be given once the new workflow is merged to develop.

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: Documentation
+on:
+  push:
+    branches: [release/*, develop]
+  pull_request:
+    branches: [release/*, develop]
```
Comment (kuznetsss) [inline .github/workflows/docs.yml]: We don't want to update documentation on new PRs.

File: docs/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.16.3)
+project(docs)
```
Comment (kuznetsss) [inline docs/CMakeLists.txt:2]: Maybe clio_docs?

Comment (godexsoft) [reply docs/CMakeLists.txt:2]: I like the short `docs` better - it's obvious it's for Clio as it's part of our repo and folder structure

File: .github/workflows/docs.yml (hunk 1 above)
Comment (godexsoft) [reply .github/workflows/docs.yml]: Yep, true

**Review by kuznetsss — APPROVED** [review]
👍

**Pull request merged**